packages/jest-cli/src/runTest.js

```diff
@@ -13,74 +13,91 @@ import type {Path, Config} from 'types/Config';
 import type {TestResult} from 'types/TestResult';
 import type {Resolver} from 'types/Resolve';
 
+const fs = require('fs');
+const {getTestEnvironment} = require('jest-config');
 const BufferedConsole = require('./lib/BufferedConsole');
+const promisify = require('./lib/promisify');
 const {
   Console,
   NullConsole,
   setGlobal,
 } = require('jest-util');
 const getConsoleOutput = require('./reporters/getConsoleOutput');
 
+const testEnvironmentPragmaRegex = /@testEnvironment\s+(.*)/;
```

Comment:
There is a docblock parser in jest-haste-map. Please let's use that instead. Feel free to extract it into a separate package "jest-docblock".